app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -11,6 +20,7 @@ import java.lang.IllegalArgumentException
 open class FeedViewModel : ViewModel() {
     private val isLoading = MutableLiveData<Boolean>()
     private val isEmpty = MutableLiveData<Boolean>()
+    private lateinit var feedRepository:FeedRepository
```

Comment:
`FeedRepository` is supposed to have a longer lifetime than `FeedViewModel`.
So instead of creating it, please inject `FeedRepository`.
In a real app we would use a fancy Dependency Injection tool for this. For this exercise you can do it in `FeedViewModelFactory`.